cmd/catalyst-node/catalyst-node.go

```diff
@@ -827,32 +827,35 @@ func (ac *PlaybackAccessControl) handleUserNew(payload []byte) []byte {
 	}
 
 	playbackID := lines[0]
-	token := requestURL.Query().Get("token")
+	jwtToken := requestURL.Query().Get("jwt")
 
 	var pubKey string
-	if token != "" {
-		claims, err := decodeJwt(token)
-
+	if jwtToken != "" {
+		claims, err := decodeJwt(jwtToken)
 		if err != nil {
 			glog.Errorf("Unable to decode JWT token %v", err)
 			return []byte("false")
 		}
 
-		if playbackID != claims.(jwt.MapClaims)["sub"].(string) {
+		if err := claims.Valid(); err != nil {
+			glog.Errorf("Invalid token %v", err)
+			return []byte("false")
+		}
+
+		if playbackID != claims.Subject {
 			glog.Errorf("PlaybackId mismatch")
 			return []byte("false")
 		}
 
-		pubKey = claims.(jwt.MapClaims)["pub"].(string)
-		expiration := time.Unix(int64(claims.(jwt.MapClaims)["exp"].(float64)), 0)
+		pubKey = claims.PublicKey
 
-		if time.Now().After(expiration) {
-			glog.Errorf("Token expired %v", expiration)
+		if time.Now().After(claims.ExpiresAt.Time) {
+			glog.Errorf("Token expired %v", claims.ExpiresAt)
 			return []byte("false")
 		}
```

Comment:
This is already checked by `jwt.RegisteredClaims.Valid()`! No need to have it here as well as far as I could tell.